src/Components/Pages/Calculator.js

```diff
@@ -1,13 +1,208 @@
-import PropTypes from 'prop-types';
+import PropTypes from 'prop-types'
 import React from 'react'
+import WASMWrapper from '../../Backend/WASMWrapper'
+
+import CalculatorSettings from '../CalculatorHelpers/CalculatorSettings'
+import CalculatorInput from '../CalculatorHelpers/CalculatorInput'
+import CalculatorOutput from '../CalculatorHelpers/CalculatorOutput'
+
+import * as convert from 'convert-units'
+
+const Units = {
+    Inputs: {
+        MeVPerNucleum: 'MeV/nucl',
+        //        Mev: 'Mev',
+    },
+    Outputs: {
+        SmallScale: 'keV/μm',
+        LargeScale: 'MeV/cm',
+        MassStoppingPower: 'Mev*g/cm^2'
+
+    }
+}
+
+const OperationMode = {
+    Dynamic: 0,
+    Performance: 1,
+}
+
 
 class CalculatorComponent extends React.Component {
     static propTypes = {
         ready: PropTypes.bool.isRequired
     }
 
+    async componentDidMount() {
+        try {
+            const [programs, ions, materials] = await Promise.all([
+                this.wrapper.getPrograms(),
+                this.wrapper.getIons(this.state.program),
+                this.wrapper.getMaterials(this.state.program)
+            ])
+            const program = programs[0]
+            const material = materials[0]
+            const ion = ions[0]
+            this.setState({ programs, ions, materials, program, material, ion })
+        } catch (err) {
+            console.log(err)
+        }
+    }
+
+    async onProgramChange(newProgram) {
+        const programNumber = (Number)(newProgram.target.value)
+        try {
+            const [materials, ions] = await Promise.all([
+                this.wrapper.getMaterials(programNumber),
+                this.wrapper.getIons(programNumber)]
+            )
+            const material = materials[0]
+            const ion = ions[0]
```

Comment:
Why do we select element with index 0 ?

Reply:
This ensures the ion and material displayed in the app exist in the newly selected program. One thing to consider here would be trying to find the currenctly selected values before resolving to this default

Reply:
Does it mean that if user changes program, then the currently selected values of material and ion are being reverted to default (being ions[0], materials[0]) ?

Reply:
It has been changed. Now it will only revert when chosen ion/material isn't available in that program.

Reply:
Very good !